openshift/system/config/redis.yml

```diff
@@ -5,3 +5,12 @@ production:
   sentinels: "<%= ENV['REDIS_SENTINEL_HOSTS'] %>"
   name: <%= ENV['REDIS_SENTINEL_NAME'] %>
   role: <%= ENV['REDIS_SENTINEL_ROLE'] %>
+  # == ACL Params ==
+  username: <%= ENV['REDIS_ACL_USERNAME'] %>
+  password: <%= ENV['REDIS_ACL_PASSWORD'] %>
+  # == TLS Params ==
+  ssl_params:
+    ca_file: <%= ENV['REDIS_TLS_CA_FILE'] %>
```

Comment:
I thought I commented somewhere on this. I would like first to know how will customers be providing their CA (perhaps depends on operator) before finalizing this variable. To make sure that this is the expected way it should work. At the very least I think we should have a default location to search for this so that users can only mount a secret and then the files are present, porta to use them.

The point about default location applies to the cert and public key variables as well.

Reply:
Yes, you mentioned that in the Apisonator PR. If not provided, Redis will verify the server certificate against all trusted certificates in the OS default ssl certs folder. In Fedora that's `/etc/ssl/certs`.

I would say the `ca_path` param would work out of the box (See [`hiredis-client`](https://github.com/redis-rb/redis-client/blob/v0.17.0/hiredis-client/lib/redis_client/hiredis_connection.rb#L24-L30)) to provide an alternative directory to `/etc/ssl/certs`. I can add a new env variable for it, so let the clients use `ca_path` or `ca_file` as they want.

AFAIK there's no equivalent config param for client certs and keys folder, but I don't think it's really needed, in OpenShift you can mount folders as well as single files.

Reply:
I pushed some changes to allow looking for trusted CA certs through `ca_path`. However, you can't just store your cert file under `ca_path`, that doesn't work (check https://github.com/redis/redis/issues/7392). You need to manually install your certificate in your OS using the official way. For Fedora, that's [update-ca-trust](https://docs.fedoraproject.org/en-US/quick-docs/using-shared-system-certificates/). And, since this mechanism essentially stores the cert in the proper format in the default folder (`/etc/ssl/certs` for Fedora), there's no difference between providing or not the `ca_path` param.

I also tried setting `ca_path` to a directory under my `$HOME` which is a symbolic link to `/etc/ssl/certs`, and it worked fine.

Summarizing, I don't think `ca_path` is much useful but since it's an officially accepted parameter and I already added the code, I think it's OK to provide it, It might be useful to somebody.

Reply:
I think we are not on the same page here. I'm asking what will be the documented way to mount the certs/key in OpenShift. I know how to create a ca path (sorry for writing ca dir previously) with `openssl-rehash`. But probably that wont be the easiest option to mange as an openshift secret. Also the documented way will show what default location to use.

Reply:
IMO if we provide both `ca_file` and `ca_path` then we are providing all available options for the operator team to define where to place the files or how to mount the secrets.

Reply:
I was thinking whether it might be decided to provide CA file as content in an env variable... unlikely.

Reply:
I'm not sure the `redis-client` gem would accept that.

Reply:
Yeah, this is fine. I think we need to check a default location though. Otherwise customers will need to both - mount the file somehow and then set the variable. It makes sense to let them only mount the file and things to start working.

Reply:
Do you suggest a path?

Reply:
I don't know off the top of my head. Need to check other secrets that we mount.

Reply:
I've checked SaaS, also our dev cluster, and it seems secrets for porta are mounted at `EXTRA_CONFIGS_DIR`, which defaults to `/opt/system-extra-configs`. See https://github.com/3scale/porta/blob/master/openshift/system/entrypoint.sh#L5

What if we use the same path `$EXTRA_CONFIGS_DIR/ca_cert.pem` ?

So if no `ca_file` nor `ca_path` are provided, we can look for `$EXTRA_CONFIGS_DIR/ca_cert.pem` and use it if present.

Reply:
Sounds good to. Probably filename should be `ca_redis.pem` or something like that. 

Do we have documented that users can edit this secret?

Reply:
> Sounds good to. Probably filename should be ca_redis.pem or something like that.

I think `ca_cert.pem` is better because this doesn't need to be exclusive for Redis. We provide a way for the user to define in which CAs they want porta to trust, it can be for Redis or anything else in the future. `ca_cert.pem` doesn't need to be a single certificate, a certificate bundle is also accepted.

> Do we have documented that users can edit this secret?

This is a new feature, not documented.

Reply:
I added this in https://github.com/3scale/porta/pull/3572/commits/81762db80a0b4e4cbfa0c5d2402b6836bcd825d2.

I finally took `config/` as the default path for the certificate. It turns out than `EXTRA_CONFIGS_DIR` is just a workaround to create symbolic links in `config/` (check the [code](https://github.com/3scale/porta/blob/master/openshift/system/entrypoint.sh)). So whatever secret the user mounts on `EXTRA_CONFIGS_DIR` will be available from `config/`.

Reply:
ok, nice. Maybe test that this works, will also help write instructions how to do by customers.

Reply:
I updated the Jira issue with some comments